bin/review-rot

```diff
@@ -71,23 +56,134 @@ def main(state_, value, duration, ssl_verify=None):
+            backup_path = path_to_config + '.backup'
+            log.info("Creating back up at " + backup_path)
+            copyfile(path_to_config, backup_path)
+            log.info("Rewriting %r in new format!" % path_to_config)
+            with open(path_to_config, 'w') as f:
+                f.write(yaml.dump(config, default_flow_style=False))
+                f.write("\narguments:\n")
+                f.close()
+
+    return config
+
+
+def load_ordered_config(path_to_config):
+    """
+      Loads the configuration in the same order as it's defined in yaml file.
+      Args:
+            path_to_config (str): Path to configuration file
+      Returns:
+            config(dict): Returns the configurations in the defined ordered
+    """
+    _mapping_tag = yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG
+
+    def dict_representer(dumper, data):
+        return dumper.represent_mapping(_mapping_tag, data.iteritems())
+
+    def dict_constructor(loader, node):
+        return collections.OrderedDict(loader.construct_pairs(node))
+
+    yaml.add_representer(collections.OrderedDict, dict_representer)
+    yaml.add_constructor(_mapping_tag, dict_constructor)
+    # read input from home directory for pull requests
+    with open(path_to_config, 'r') as f:
+        config = yaml.load(f)
+    return config
+
+
+def get_arguments(config_arguments):
+    """
+       Get the arguments provided in configuration file
+       and command line arguments
+       Args:
+             config_arguments (dict): Arguments specified in yaml file
+       Returns:
+             arguments (dict): Returns the parsed arguments
+     """
+    parsed_arguments = {}
+    command_line_args = vars(args)
+    grouped_arguments = {'state', 'duration', 'value'}
+    logged_error = False
+
+    for arg in command_line_args:
+        if command_line_args.get(arg) is not None:
+            parsed_arguments[arg] = command_line_args.get(arg)
+
+    parsed_arguments['ssl_verify'] = False if args.insecure else args.cacert
+
+    if config_arguments is not None:
+        for argument in config_arguments:
+            if argument not in command_line_args.keys():
+                # if key is present in grouped_arguments,
+                # all the associated arguments should also
+                # be specified in ~/.reviewrot.yaml file
+                if argument not in grouped_arguments or \
+                        (argument in grouped_arguments and
+                         grouped_arguments.issubset(config_arguments.keys())):
+                    parsed_arguments[argument] = config_arguments[argument]
+                elif not logged_error:
+                    log.warn("Either no or all arguments (state, duration and"
+                             " value) are required in ~/.reviewrot.yaml file")
+                    logged_error = True
+
+        if config_arguments.get('debug'):
+            log.setLevel(level=logging.DEBUG)
```

Comment:
Why is this option treated differently than the rest? I would expect to be able to specify debug from the cli and have it override the value from the config. Also, to keep this function single purpose, it would be best to set a key/value pair in `parsed_arguments` rather than actually setting the log level here.